### PR TITLE
[DUOS-256][risk=no] Build/push to GCR action

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
Partially addresses https://broadworkbench.atlassian.net/browse/DUOS-256

* Adds new build action using secrets pushed from TF
* ~Requires https://github.com/broadinstitute/terraform-ap-deployments/pull/452~ Done
* ~Requires disabling the image build/push/gke-deploy portions of the [`consent-build`](https://fc-jenkins.dsp-techops.broadinstitute.org/view/Build/job/consent-build/) jenkins job just prior to merging~ Done
* See https://github.com/DataBiosphere/duos-ui/blob/develop/.github/workflows/build.yaml for reference

This is verified with the [most recent action run](https://github.com/DataBiosphere/consent/runs/4033598785?check_suite_focus=true). New image with this build has been pushed to GCR.

### Action Results:
Screen Shot 2021-10-28 at 7 10 00 AM

### GCR image with tags:
Screen Shot 2021-10-28 at 7 11 27 AM

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by rushtong** [comment]
Some failing tests are very unrelated to this PR. Will address separately, they are transient.
```
Error:  Errors: 
Error:    UserServiceTest.testCreateUserDuplicateEmail »  Unexpected exception, expected...
Error:    UserServiceTest.testGetUsersAsRoleAdmin:379 » IllegalArgument duplicate elemen...
```

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+name: Tag, Build and Push Image
+
+on:
+  push:
+    branches:
+    - develop
+    paths-ignore:
+    - 'CONTRIBUTING.md'
```
Comment (JVThomas) [inline .github/workflows/build.yaml:8]: Should we include LICENSE.txt as well?

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    - 'DEVNOTES.md'
+    - 'README.md'
+    - '.github/**'
+  pull_request:
+    branches:
+    - develop
+    paths-ignore:
+    - 'CONTRIBUTING.md'
```
Comment (JVThomas) [inline .github/workflows/build.yaml:17]: Same LICENSE.txt question as above

File: .github/workflows/build.yaml (hunk 1 above)
Comment (rushtong) [reply .github/workflows/build.yaml:8]: I was about to comment that we don't do that in the [UI build](https://github.com/DataBiosphere/duos-ui/blob/develop/.github/workflows/build.yaml) ... but then realized that we don't have a `LICENSE.txt` file at all there. Yes, will add that as an exclusion and file a ticket to add a license to duos-ui.

**Review by JVThomas — APPROVED** [review]
Looks good!

**Pull request merged**